src/server/cea_608_encoder/caption_string_utility.py

```diff
@@ -107,17 +107,33 @@ def create_byte_pairs_for_caption_string(caption_string: str, channel_toggle: in
     :return: list of byte pairs
     """
     byte_list = []
+    # for determining whether or not a 0x80 needs to be appended before the end
+    # of the string or before an extended character set. if it is odd, we need
+    # to append before putting the extended char header.
+    basic_chars_is_odd = False
     for letter in caption_string:
         char_set_name = which_char_set(letter)
         first_byte = which_channel(channel_toggle,char_set_name)
         if first_byte != 0x00:
+            # check to see if the basic chars prior to this was an odd number,
+            # and if so, append 0x80
+            if basic_chars_is_odd:
+                byte_list.append(hex(get_single_null_byte_with_parity()))
             if check_parity(first_byte) == 0:
                 first_byte = add_parity_to_byte(first_byte)
             byte_list.append(hex(first_byte))
+            # since we know we just appended a full bytepair, we set to false.
+            basic_chars_is_odd = False
+        else:
+            basic_chars_is_odd = ~basic_chars_is_odd
```

Comment:
i am 95% sure this is not doing what i think it should be doing

Reply:
if you're trying to negate basic_chars that is

Reply:
i tested it in a shell, it actually does work just fine- when i print after negating, it turns false into 0 and true into 1. i can totally change it

Reply:
i thought this would be best for readability, didnt know it would be weird.

Reply:
![Screen Shot 2019-10-31 at 6 03 19 PM](https://user-images.githubusercontent.com/22465473/67995656-d1b8d500-fc08-11e9-997b-dc17aa91e97f.png)